assets/pages/contributions-landing/components/contributionsBundle.jsx

```diff
@@ -109,8 +109,29 @@ const contentText = {
   ),
 };
 
+const upsellRecurringContributionsTestContentText = {
+  control: defaultContentText.USD,
+  benefitsOfBoth: (
+    <span>
+      Make a monthly commitment to support The Guardian long-term or a one-time contribution
+      as and when you feel like it &ndash; choose the option that suits you best.
+    </span>
+  ),
+  shorterControl: (
+    <span>
+      If you’re able, please consider <strong>monthly</strong> support &ndash;
+      it will help to fund The Guardian’s journalism for the long-term.
+    </span>
+  ),
+};
+
+function getContentText(props: PropTypes) {
+  return upsellRecurringContributionsTestContentText[props.abTests.upsellRecurringContributions] ||
+    defaultContentText[props.currency.iso];
```

Comment:
Can I just double-check my understanding of what happens here if you're not in the US?

You won't be assigned to this test, and therefore the lookup for `upsellRecurringContributions` in `props.abTests` will return `undefined`? This means that the lookup for the key `undefined` in `upsellRecurringContributionsTestContentText` will also return `undefined`? Which in turn means that the `||` expression will treat that `undefined` as falsy, and move to `defaultContentText[props.currency.iso]` expression?

If that sounds correct, I'm fine with it because it works, I'm just wondering if there's a way that what's happening here could be slightly clearer?

Reply:
Yes - if they're in the test, serve the copy for that test, else revert to the default copy for the given user. I don't know javascript too well - would an if/else statement be clearer / more idiomatic?

Reply:
Perhaps something like?:

```js

const upsellRecurring = props.abTests.upsellRecurringContributions;

if (upsellRecurring && upsellRecurring !== 'notintest') {
  return upsellRecurringContributionsTestContentText[upsellRecurring];
}

return defaultContentText[props.currency.iso];
```

Again, what exists currently works fine, it's just a suggestion.

Reply:
sure - thanks

Reply:
Or, if you're worried that somehow `props.abTests.upsellRecurringContributions` could end up with a random value that's not one of the variant names:

```js

const upsellRecurring = props.abTests.upsellRecurringContributions;
let contentText = '';

if (upsellRecurring && upsellRecurring !== 'notintest') {
  contentText = upsellRecurringContributionsTestContentText[upsellRecurring];
}

return contextText || defaultContentText[props.currency.iso];
```

Although I would have though that was unlikely. I wonder if there's any way we can get type safety on the variant strings? @svillafe any thoughts?